apps/application-system/api/src/app/modules/application/application.controller.ts

```diff
@@ -558,6 +560,8 @@ export class ApplicationController {
   ): Promise<ApplicationResponseDto> {
     const { type } = input
 
+    this.fileService.validateApplicationType(application.typeId)
```

Comment:
Why not put the guards inside the corresponding functions? 😶😊

Reply:
I actually started that way and then I thought about who should have this responsibility and decided on this should be on a higher level. Up for discussion imo 😄